src/allmydata/watchdog/inotify.py

```diff
@@ -0,0 +1,139 @@
+    IN_WATCH_MASK, IN_ACCESS, IN_MODIFY, IN_ATTRIB, IN_CLOSE_NOWRITE, IN_CLOSE_WRITE, \
+    IN_OPEN, IN_MOVED_FROM, IN_MOVED_TO, IN_CREATE, IN_DELETE, IN_DELETE_SELF, \
+    IN_MOVE_SELF, IN_UNMOUNT, IN_Q_OVERFLOW, IN_IGNORED, IN_ONLYDIR, IN_DONT_FOLLOW, \
+    IN_MASK_ADD, IN_ISDIR, IN_ONESHOT, IN_CLOSE, IN_MOVED, IN_CHANGED
+[humanReadableMask, \
+    IN_WATCH_MASK, IN_ACCESS, IN_MODIFY, IN_ATTRIB, IN_CLOSE_NOWRITE, IN_CLOSE_WRITE, \
+    IN_OPEN, IN_MOVED_FROM, IN_MOVED_TO, IN_CREATE, IN_DELETE, IN_DELETE_SELF, \
+    IN_MOVE_SELF, IN_UNMOUNT, IN_Q_OVERFLOW, IN_IGNORED, IN_ONLYDIR, IN_DONT_FOLLOW, \
+    IN_MASK_ADD, IN_ISDIR, IN_ONESHOT, IN_CLOSE, IN_MOVED, IN_CHANGED]
+
+
+TRUE  = 0
+FALSE = 1
+
+NOT_STARTED = "NOT_STARTED"
+STARTED     = "STARTED"
+STOPPING    = "STOPPING"
+STOPPED     = "STOPPED"
+
+class INotifyEventHandler(FileSystemEventHandler):
+
+    def __init__(self, path, mask, callbacks, pending_delay):
+        print "init INotifyEventHandler"
+        FileSystemEventHandler.__init__(self)
+        self._path = path
+        self._mask = mask
+        self._callbacks = callbacks
+        self._pending_delay = pending_delay
+        self._pending = set()
+
+    def process(self, event):
+        event_filepath_u = event.src_path.decode(encodingutil.get_filesystem_encoding())
+
+        if event_filepath_u == unicode_from_filepath(self._path):
+            # ignore events for parent directory
+            return
+        #try:
+        #    event_path = self._path.preauthChild(event.src_path)
+        #except InsecurePath, e:
+        #    print "failed: %r" % (e,)
+        #    return
+
+        def _maybe_notify(path):
+            if path in self._pending:
+                return
+            self._pending.add(path)
+            def _do_callbacks():
+                print "DO CALLBACKS"
+                self._pending.remove(path)
+                event_mask = IN_CHANGED
+                if event.is_directory:
+                    event_mask = event_mask | IN_ISDIR
+                for cb in self._callbacks:
+                    try:
+                        cb(None, FilePath(path), event_mask)
+                    except Exception, e:
+                        log.err(e)
+            _do_callbacks()
+        reactor.callFromThread(_maybe_notify, event_filepath_u)
+
+    def on_any_event(self, event):
+        print "PROCESS EVENT %r" % (event,)
+        self.process(event)
+
+class INotify(PollMixin):
+    """
+    I am a prototype INotify, made to work on Mac OS X (Darwin)
+    using the Watchdog python library. This is actually a simplified subset
+    of the twisted Linux INotify class because we do not utilize the watch mask
+    and only implement the following methods:
+     - watch
+     - startReading
+     - stopReading
+     - wait_until_stopped
+     - set_pending_delay
+    """
+    def __init__(self):
+        self._pending_delay = 1.0
+        self.recursive_includes_new_subdirectories = False
```

Comment:
yes and we aren't using the watchdog module for windows. so the question is should it be set to true for mac osx and BSD platforms?